src/Client.js

```diff
@@ -114,6 +114,7 @@ class RPCClient extends BaseClient {
         resolve(this);
       });
       this.transport.once('close', reject);
+      this.transport.once('error', reject);
       this.transport.connect({ client_id: this.clientID });
```

Comment:
just do `this.transport.connect(...).catch(reject)` instead wrapping the entire thing below

Reply:
Thats a good point! Changing that!